Doc/tutorial/errors.rst

```diff
@@ -496,3 +496,136 @@ used in a way that ensures they are always cleaned up promptly and correctly. ::
 After the statement is executed, the file *f* is always closed, even if a
 problem was encountered while processing the lines. Objects which, like files,
 provide predefined clean-up actions will indicate this in their documentation.
+
+
+.. _tut-exception-groups:
+
+Raising and Handling Multiple Unrelated Exceptions
+==================================================
+
+There are situations where it is necessary to report several exceptions that
+have occurred. This it often the case in concurrency frameworks, when several
+tasks may have failed in parallel, but there are also other use cases where
+it is desirable to continue execution and collect multiple errors rather than
+raise the first exception.
+
+The builtin :exc:`ExceptionGroup` wraps a list of exceptions so that they can
+be raised together. It is an exception itself, so it can be caught like any
+other exception. ::
+
+   >>> def f():
+   ...     excs = [OSError('error 1'), SystemError('error 2')]
```

Comment:
Python allows raising exceptions without creating an instance of it as interpreter handles it automatically:
```py

try:
    raise ValueError
except ValueError:
    print('raised ValueError')
```
I would expect that `excs = [ValueError]` to work here just fine but it seems that's not the case:

```py
e = ExceptionGroup("Group 1", [ValueError])
```
Errors out as 

```console
Running Debug|x64 interpreter...
Traceback (most recent call last):
  File "D:\cpython\main.py", line 1, in <module>
    e = ExceptionGroup("Group 1", [ValueError])
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: Item 0 of second argument (exceptions) is not an exception
```

Reply:
Is this behavior expected in `ExceptionGroup` ?

Reply:
Yes. It has to be an exception instance.

Reply:
When you raise an exception by type only, an instance is implicitly created by the "raise" instruction. This syntax exists for historical reasons (in the past you could raise more things than you can now) but it doesn't really add any power to the language.

Reply:
It adds not needing to type `()` so it _has_ been an idiom used in some code where adding a parameter or message to the error doesn't seem necessary. The construction is implied by Python. The stdlib has hundreds of these:

```
raise StopIteration
raise StopAsyncIteration
raise IndexError
raise EOFError
raise NotImplementedError
```

If we are intentionally not supporting this Python idiom in exception groups we should explicitly document that.  Though it doesn't belong in the tutorial, probably up in exceptions.rst.

It sounds like an easy thing to implement in the BaseExceptionGroup construction code.  if a type is passed and it inherits from exception, instantiate it.  This would make the feature more consistent with the rest of Python.  Is there some reason we decided not to?

Reply:
I can’t imagine a use case for this. You would typically wrap in a group exceptions that you caught.

Of course, we can support it if there’s a reason to.

Reply:
Historically speaking, the instantiation-less syntax was originally much faster when the exception was caught in C.  IIRC, the last time this was analyzed was in the 2.5 time frame.  That said, that use case doesn't translate to EGs and I agree that I can't think of a use case for them.  Still, +1 for documenting this difference.  (...ah, I see later on that you have)